lib/github/oauth.js

```diff
@@ -73,14 +73,15 @@ module.exports = (robot) => {
       token: accessToken,
     });
     const user = (await github.users.get({})).data;
+    const defaults = { accessToken, login: user.login };
 
     // store github user id and connect to slack user id
     const [gitHubUser, created] = await GitHubUser.findOrCreate({
       where: { id: user.id },
-      defaults: { accessToken },
+      defaults,
```

Comment:
Hey @gimenete 👋 ,

Just wanted to make sure my assumption is correct here and that both values inside of `default` have to match in order to return a user object. Either way, should be good from a security perspective as this is the result of authenticating against GitHub to retrieve user details.

Reply:
Only the fields in the `where` have to match in order to find the user. If it's not found, a user is created with the attributes resulting of merging the `where` and the `default` ones. In the next sentence there's an `if` to update the user with the values in `defaults` if the user was found (not created).

For additional information, the `user` object comes from the GitHub API so we know that the `id` is unique and never changes.

Reply:
Thanks @gimenete - I was just trying to figure out if there would be any ramifications if the one piece of information under user control (`user.login` - since they can change their login information) would be impacted. For example, a scenario where:

* 1 - The `acme` org changes their login to `theacme`
* 2 - A user takes advantage of this login name change and takes the old org name `acme`

I suppose it wouldn't matter as we **first** perform the lookup by the `id` which doesn't change as you'd mentioned.

Reply:
We are storing the `login` just for debugging purposes and we are always querying by `id` and primary keys are `ids` too. Just to make that clear, as @wilhelmklopp suggested it, I've also added the following comment to the new columns that store login/names:

> Storing only for debugging purposes. Do not use for querying data since it might change

Reply:
Perfect 👌 ! Thank you @gimenete